.github/workflows/deploy.yml

```diff
@@ -51,13 +51,44 @@ jobs:
           echo "Tag will not be created."
         fi
 
+  plan_builds:
+    name: Plan builds
+    runs-on: ubuntu-latest
+    outputs:
+      download_linux_arm32_runid: ${{ steps.plan_builds.outputs.download_linux_arm32_runid }}
+      download_linux_arm64_runid: ${{ steps.plan_builds.outputs.download_linux_arm64_runid }}
+      download_linux_x64_runid: ${{ steps.plan_builds.outputs.download_linux_x64_runid }}
+      download_macos_runid: ${{ steps.plan_builds.outputs.download_macos_runid }}
+      download_windows_x64_runid: ${{ steps.plan_builds.outputs.download_windows_x64_runid }}
+      download_windows_portable_runid: ${{ steps.plan_builds.outputs.download_windows_portable_runid }}
+      platforms_to_build: ${{ steps.plan_builds.outputs.platforms_to_build }}
+    steps:
+      - id: plan_builds
+        name: Plan builds
+        env:
+          PLATFORMS: ${{ inputs.platforms }}
+        run: |
+          platform_regex='(linux_arm32|linux_arm64|linux_x64|macos|windows_x64|windows_portable)'
+          int_regex="(0|[1-9][0-9]*)"
+
+          platforms_to_build=()
+          for platform in ${PLATFORMS}; do
+            if [[ "${platform}" =~ "${platform_regex}@${int_regex}" ]]; then
+              echo "download_${BASH_REMATCH[1]}_runid=${BASH_REMATCH[2]}" | tee -a "${GITHUB_OUTPUT}"
```

Comment:
Thanks for introducing me to [`BASH_REMATCH`](https://medium.com/@linuxadminhacks/understanding-the-bash-rematch-in-bash-183a5b065081)! That looks super useful!